.pre-commit-config.yaml

```
@@ -0,0 +1,12 @@
+# See https://pre-commit.com for more information
+repos:
+  - repo: https://github.com/PyCQA/flake8
+    rev: 6.0.0
+    hooks:
+      - id: flake8
+        # Required to make flake8 read from pyproject.toml for now :(
+        additional_dependencies: ["flake8-pyproject"]
+  - repo: https://github.com/psf/black
+    rev: 23.3.0
```

```suggestion
    rev: 23.7.0
```
I know it doesn't change much, but may as well be up to date when added.